java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -185,8 +187,7 @@ public ProgramState apply(ProgramState input) {
         }
       });
 
-      if (variableTree.symbol().metadata().isAnnotatedWith("javax.annotation.CheckForNull")
-        || variableTree.symbol().metadata().isAnnotatedWith("javax.annotation.Nullable")) {
+      if (isEqualsMethod || parameterCanBeNull(variableTree)) {
```

Comment:
not usually found of downcasting but in this case it is better than redefining the way to detect equals method: You can use : https://github.com/SonarSource/sonar-java/blob/master/java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java#L326-326 by downcasting to MethodTreeImpl in this case : 
`((MethodTreeImpl) tree).isEqualsMethod`